### PR TITLE
Cole db

### PR DESCRIPTION
**Pull request by ColePurboo**
updated UserDb (addUser, validateUser etc)

### CODE REVIEW

**Review by jaylovegood — DISMISSED** [review]
Looks great and ready to merge!

File: src/main/java/business/rules/dbs/UserDB.java [hunk 1]
```diff
@@ -1,27 +1,99 @@
...
+import java.util.Map;
+
+// Need one static function to load database from the local system (from computer objectreader/object writer)
+// needs to be serializable
+// reading from a file and typecast to the userDB type
+// functionality : add user, remove user, modify user, update name, update password
+// see if the user has a file already, if not, make a new one, if they do then you get the location and use as input stream
+// thread/ asynch /every time
+
+public class UserDB implements Serializable {
```
Comment (jaylovegood) [inline src/main/java/business/rules/dbs/UserDB.java]: wouldn't it be better to make it implement DB as well?

**Review by jaylovegood — DISMISSED** [review]
Looks good!
I'm not sure how the DB interface works rn but would it be possible to make UserDB implement DB as well?

**Review by alexfigner — CHANGES_REQUESTED** [review]
I fixed the conflict in User. I echo Jisu's comment that UserDB should implement both Serializable and DB, since they are interfaces and can do so.

**Review by alexfigner — APPROVED** [review]
Great work - thanks for making the change to UserDB.

**Pull request merged**